staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -453,7 +453,7 @@ func IsValidSocketAddr(value string) []string {
 		errs = append(errs, "must be a valid socket address format, (e.g. 0.0.0.0:10254 or [::]:10254)")
 		return errs
 	}
-	portInt, _ := strconv.Atoi(port)
+	portInt, _ := netutils.ParsePort(port, false)
 	errs = append(errs, IsValidPortNum(portInt)...)
```

Comment:
this looks redundant?

Reply:
It is. I just noticed that IsValidPortNum checks for an overflow. Should I revert the change or apped the error from netutils.ParsePort to the errs